src/conversions.jl

```diff
@@ -895,6 +895,54 @@ to_3d_scale(x::VecTypes) = to_ndim(Vec3f, x, 1)
 to_3d_scale(x::AbstractVector) = to_3d_scale.(x)
 
 
+## UV Transforms
+
+convert_attribute(x::Vector, k::key"uv_transform") = convert_attribute.(x, (k,))
+convert_attribute(x, k::key"uv_transform") = convert_attribute(uv_transform(x), k)
+convert_attribute(x::Mat3f, k::key"uv_transform") = x[Vec(1,2), Vec(1,2,3)]
+convert_attribute(x::Mat{2, 3, Float32}, k::key"uv_transform") = x
+convert_attribute(x::Nothing, k::key"uv_transform") = x
+
+# these return Mat3f's so they can be multiplied
+uv_transform(packed::Tuple) = uv_transform(packed...)
+# prefer scale as single argument since it may be useful for patterns
+# while just translation is mostly useless
+uv_transform(scale::VecTypes{2, <: Real}) = uv_transform(Vec2f(0), scale)
+function uv_transform(translation::VecTypes{2, <: Real}, scale::VecTypes{2, <: Real})
+    return Mat3f(
+        scale[1], 0, 0,
+        0, scale[2], 0,
+        translation[1], translation[2], 1
+    )
+end
+function uv_transform(translation::VecTypes{2, <: Real}, scale::VecTypes{2, <: Real}, angle::Real)
+    return Mat3f(
+         scale[1] * cos(angle), scale[2] * sin(anlge), 0,
+        -scale[1] * sin(angle), scale[2] * cos(angle), 0,
+        translation[1], translation[2],                1
+    )
+end
+function uv_transform(angle::Real)
+    return Mat3f(
+         cos(angle), sin(anlge), 0,
+        -sin(angle), cos(angle), 0,
+        0, 0, 1
+    )
+end
+function uv_transform(action::Symbol)
+    # TODO: do some explicitly named operations
+    # TODO: check these
+    if action == :rotr90
+        return uv_transform(pi/2)
+    elseif action == :rotl90
+        return uv_transform(-pi/2)
+    elseif action == :swap_xy
+        return Mat3f(0,1,0, 1,0,0, 0,0,1)
+    else
+        error("Transformation :$action not recognized.")
+    end
+end
```

Comment:
My current setup for the conversion pipeline allows:
- direct `Mat{2, 3, Float32}`
- `Mat3f` because these are easier to combine
- via `uv_transform()`:
  - `scale::VecTypes{2, <: Real}`
  - `(translation::VecTypes{2, <: Real}, scale::VecTypes{2, <: Real)`
  - `rotation::Real` (2D rotation, single angle in radians)
  - `(translation, scale, rotation)`
  - `named::Symbol`
- vectors of any of the above
- `nothing` which should remove the matmult in shaders

I'm wondering if 
- `uv_transform` should just be a method of `transformationmatrix()`
- we should allow a tuple of multiple symbols which then combine operations (e.g. `(:flip_x, :rotr90)` combining to do both) but I'm worried that the order of operations will be confusing

Also feel free to suggest some more named operations.

Reply:
Maybe also something like `transpose`?  Given how flexible this is, it would be pretty easy to have a series of structs with stored values (e.g. `RestrictTo(bbox)` if you want to use only some subsection of a texture, etc), which you then implement `uv_transform(::RestrictTo)` for.  Is that understanding of the pipeline correct?

Reply:
I added transpose.
Instead of `RectrictTo` we could just use `Rect2f`` directly. Though you can also use (translation, scale) already.

> uv_transform should just be a method of transformationmatrix()

No, because some of the transformations assume $0 \le uv \le 1$ and don't make sense in a general matrix context